packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -9,115 +9,153 @@ import 'package:flutter/scheduler.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter/widgets.dart';
 
-import 'colors.dart';
-import 'constants.dart';
 import 'debug.dart';
-import 'icon_button.dart';
-import 'icons.dart';
-import 'material.dart';
 import 'material_localizations.dart';
-import 'text_button.dart';
 import 'text_selection_theme.dart';
+import 'text_selection_toolbar.dart';
+import 'text_selection_toolbar_text_button.dart';
 import 'theme.dart';
 
 const double _kHandleSize = 22.0;
 
-// Minimal padding from all edges of the selection toolbar to all edges of the
-// viewport.
-const double _kToolbarScreenPadding = 8.0;
-const double _kToolbarHeight = 44.0;
-// Padding when positioning toolbar below selection.
+// Padding between the toolbar and the anchor.
 const double _kToolbarContentDistanceBelow = _kHandleSize - 2.0;
 const double _kToolbarContentDistance = 8.0;
 
-/// Manages a copy/paste text selection toolbar.
-class _TextSelectionToolbar extends StatefulWidget {
-  const _TextSelectionToolbar({
+/// Android Material styled text selection controls.
+class MaterialTextSelectionControls extends TextSelectionControls {
+  /// Returns the size of the Material handle.
+  @override
+  Size getHandleSize(double textLineHeight) => const Size(_kHandleSize, _kHandleSize);
+
+  /// Builder for material-style copy/paste text selection toolbar.
+  @override
+  Widget buildToolbar(
```

Comment:
My approach for this PR was to not break the TextSelectionControls and buildToolbar interfaces.

The class that I'm making public is TextSelectionToolbar and not the _TextSelectionControlsToolbar class that I'm instantiating here because _TextSelectionControlsToolbar is designed for the buildToolbar API, which may be changed in the future.  I wrote TextSelectionToolbar's API to be as straightforward as possible without depending on the existing buildToolbar API.  In the future when we do a breaking refactor of TextSelectionControls and buildToolbar, TextSelectionToolbar should continue to be used without changes, and we can hopefully get rid of _TextSelectionControlsToolbar.